### ISSUE TITLE
Unable to load GIF from Google's Storage API

### ISSUE THREAD

**Issue by MayaLekova** [onevcat/Kingfisher#586]
I'm writing a [simple app](https://github.com/MayaLekova/ios-is-now-illegal) which uses Kingfisher to display gifs inside an `AnimatedImageView`. When I'm testing with some gif from giphy.com, it loads correctly. However when I try to load a gif stored on storage.googleapis.com, the library returns "bad data" error (Error Domain=com.onevcat.Kingfisher.Error Code=10000)

What I tried:

1. Checking that the [gif in question](https://storage.googleapis.com/is-now-illegal.appspot.com/gifs/TEST.gif) is public (it doesn't need authentication, loads properly in the browser)
2. Setting the "Accept-Encoding" header of the request to "sdch" (according to a suggestion from the guys maintaining the Google Storage instance)
3. Debugging a little bit into Kingfisher's code - I noticed that the HTTP status code of the response is 200 and the `frameCount` is decoded properly in the `decode` method (71 in the case of this particular gif)
4. What actually fails is the call to CGImageSourceCopyPropertiesAtIndex [here](https://github.com/onevcat/Kingfisher/blob/master/Sources/Image.swift#L271)

I went through all "gif" issues here, as well as gif-related questions in StackOverflow, but without any clue. As I'm new to iOS programming, I'm unable to investigate further what is the exact problem with the properties, so any help will be appreciated! :)

**Comment by MayaLekova** [comment]
Sorry if I'm adding too much info, but:
I switched to using a not-anymore supported library called MapleBacon and it succesfully loads the gif. They have a similar code to obtain the delays between the gif frames, but also include a fallback in case there are no delays embedded. You can take a look [here](https://github.com/zalando-incubator/MapleBacon/blob/master/Library/MapleBacon/MapleBacon/ImageGifExtension.swift#L61)

**Comment by onevcat** [comment]
@MayaLekova 

Thanks for the information. I seems to be a gif decoding issue (Kingfisher is not a gif library after all). I guess we could add the same fallback in Kingfisher. I will check your sample app above and see what is the best way to solve it.

Thanks for your time in investigating it. It is really helpful and you are great!

**Comment by onevcat** [comment]
Hi, may I get the GIF file with problem?

I tried the one you mentioned in your original post ([this](https://storage.googleapis.com/is-now-illegal.appspot.com/gifs/TEST.gif)) and it seems that Kingfisher decode it without problem for both `UIImageView` and `AnimatedImageView` (and it has 95 frames instead of 71). So I guess it is not the one with issue.

**Comment by MayaLekova** [comment]
Hello @onevcat , thank you for your reply! Really pleased if I can help.

Seems the guys hosting the gifs have changed the one that I tested with initially, but here's [another one](https://storage.googleapis.com/is-now-illegal.appspot.com/gifs/BABA.gif) constructed from the same static frames - it should contain 71 frames again. Hope this will help you!

PS: Just in case they wipe this other gif, I've committed it as part of my project [here](https://github.com/MayaLekova/ios-is-now-illegal/raw/master/BABA.gif)
